### PR TITLE
Get rid of the Union

### PR DESCRIPTION
**Pull request by cclauss**
### Describe your change:

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_101/sol1.py [hunk 1]
```diff
@@ -44,9 +44,8 @@
 from __future__ import annotations
 
 from collections.abc import Callable
-from typing import Union
 
-Matrix = list[list[Union[float, int]]]
+Matrix = list[list[Union[float | int]]]
```
Comment (poyea) [inline project_euler/problem_101/sol1.py:48]: or is it
```suggestion
Matrix = list[list[float | int]]
```

**Review by poyea — APPROVED** [review]

**Comment by cclauss** [comment]
Do you know how we can enable https://pre-commit.ci on this repo?  That would automate a lot of busy work.

**Comment by poyea** [comment]
Should we request it on the website?

**Comment by poyea** [comment]
I made a request - not sure where it will be routed to

**Pull request merged**